cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -23,6 +26,95 @@
 import re
 import uuid
 
+
+class GCPResourceType(object):
+
+    def __init__(self, platform, resource_type):
+        self.platform = platform
+        self.resource_type = resource_type
+
+    @property
+    def kind(self):
+        return '%s#%s' % (self.platform, self.resource_type)
+
+    @property
+    def url_field_name(self):
+        return '%ss' % self.resource_type
+
+
+# GCP resource types we are interested in
+FORWARDING_RULE_TYPE = GCPResourceType('compute', 'forwardingRule')
+INSTANCE_TYPE = GCPResourceType('compute', 'instance')
+NETWORK_TYPE = GCPResourceType('compute', 'network')
+REGION_TYPE = GCPResourceType('compute', 'region')
+TARGET_INSTANCE_TYPE = GCPResourceType('compute', 'targetInstance')
+
+ALL_RESOURCE_TYPES = [FORWARDING_RULE_TYPE,
+                      INSTANCE_TYPE,
+                      NETWORK_TYPE,
+                      REGION_TYPE,
+                      TARGET_INSTANCE_TYPE]
+
+
+class GCPResourceUrl(GCPResourceType):
+
+    def __init__(self, platform, resource_type, project=None, region=None,
+                 zone=None, name=None, connection=None):
+        super(GCPResourceUrl, self).__init__(platform, resource_type)
+        self.project = project
+        self.region = region
+        self.zone = zone
+        self.name = name
+        self.connection = connection
+
+    @staticmethod
+    def parse_url(url, provider=None):
+        parts = url.split('/')
+        out = None
+        for resource_type in ALL_RESOURCE_TYPES:
+            if parts[-2] == resource_type.url_field_name:
+                out = GCPResourceUrl(resource_type.platform,
+                                     resource_type.resource_type)
+        if not out:
+            return out
+        out.name = parts[-1]
+
+        # If a cloud provider is given, set the connection.
+        if provider:
+            if out.platform == 'compute':
+                out.connection = provider.gce_compute
+            elif out.platform == 'storage':
+                out.connection = provider.gcp_storage
+
+        # Set region, zone, and project fields.
```

Comment:
Please add a comment regarding what you are looking for here in the URL.
Again, an example goes a long way.